### PR TITLE
Properly handle the case where an environment variable is an empty string

### PR DESCRIPTION
**Pull request by ttung**
v.idup is undefined when called on an empty array, and currently it
returns null.  This is undesirable because we want to differentiate
between not-present and empty-but-present.

### CODE REVIEW

**Comment by JakobOvrum** [comment]
LGTM. Is there an issue report associated with this?

**Comment by ttung** [comment]
No, I just ran into it.

File: std/process.d [hunk 1]
```diff
@@ -3088,7 +3088,12 @@ private:
 
             // Cache the last call's result.
             static string lastResult;
-            if (v != lastResult) lastResult = v.idup;
+            if (v.empty) {
```
Comment (JackStouffer) [inline std/process.d]: brace style, needs to be on its own line

File: std/process.d [hunk 2]
```diff
@@ -3088,7 +3088,12 @@ private:
...
+                // dup on an empty array is undefined.
+                lastResult = "";
+            } else if (v != lastResult) {
```
Comment (JackStouffer) [inline std/process.d]: same goes for `else if`s

File: std/process.d [hunk 3]
```diff
@@ -3116,11 +3121,15 @@ unittest
     assertThrown(environment["std_process"]);
 
     // get() without default value
-    assert (environment.get("std_process") == null);
+    assert (environment.get("std_process") is null);
```
Comment (JackStouffer) [inline std/process.d]: common usage is to not have the space between `assert` and the `(`; might as well fix it while this line is being changed

**Comment by JackStouffer** [comment]
When the style is fixed LGTM

**Comment by JackStouffer** [comment]
Um, I meant fix it for that line specifically, but w/e. Thanks for putting it in a separate commit though.

LGTM

**Comment by Hackerpilot** [comment]
Auto-merge toggled on

**Comment by JackStouffer** [comment]
Why hasn't the auto tester merged this?

**Comment by Hackerpilot** [comment]
That's a good question.

**Comment by Hackerpilot** [comment]
https://auto-tester.puremagic.com/show-run.ghtml?projectid=1&runid=1913350&isPull=true

**Comment by braddr** [comment]
Auto-merge toggled off

**Comment by braddr** [comment]
Turned off auto-merge to quit prioritizing this build since it's broken on windows.

**Comment by tsbockman** [comment]
> Turned off auto-merge to quit prioritizing this build since it's broken on windows.

Possibly related: https://github.com/D-Programming-Language/druntime/pull/1404#issuecomment-171425594

**Comment by quickfur** [comment]
The autotester is still failing, and doesn't seem to be related to `timerfd`.  The line number doesn't seem to make any sense (points to `import core.thread;`??! though claiming to be a unittest failure).  Something wrong with this fix on Windows?  Does Windows allow empty environment variables or does it treat them the same as unsetting?

**Comment by quickfur** [comment]
P.S. A little googling reveals that, disturbingly(?), the Windows command prompt treats empty environment variables the same as non-existent. So this fix will not work on Windows.

Probably a `version(Posix)` is in order, somewhere in this PR?

**Pull request closed without merging**